toolchain/lower/testdata/function/generic/call_basic.carbon

```diff
@@ -179,7 +179,7 @@ fn M() {
 // CHECK:STDOUT:
 // CHECK:STDOUT: define linkonce_odr void @_CH.Main.15b1f98bd9cc0c5b(ptr sret({}) %return, ptr %x) !dbg !76 {
 // CHECK:STDOUT: entry:
-// CHECK:STDOUT:   ret ptr %x, !dbg !77
+// CHECK:STDOUT:   ret void, !dbg !77
```

Comment:
Do I understand correctly that this is the instantiation of `H` for `C`? If so, it seems like the old code was right and this is wrong, because it needs to return `x`. Am I missing something?

Reply:
`C` is returned via the `sret` parameter, which means that a pointer to the storage for a `C` object is passed in and initialized by the function, rather than being returned at the LLVM level. Note that the function's signature says its return type is `void`; the old return instruction is malformed LLVM IR.